ext/iconv/config.m4

```diff
@@ -28,20 +28,11 @@ if test "$PHP_ICONV" != "no"; then
 
     if test -z "$iconv_impl_name"; then
       AC_MSG_CHECKING([if using GNU libiconv])
-      AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <iconv.h>
-#include <stdio.h>
-int main(void) {
-  printf("%d", _libiconv_version);
-  return 0;
-}
-      ]])],[
+      AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <iconv.h>]], [[int main(){return _libiconv_version;}]])],[
```

Comment:
```suggestion
      AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <iconv.h>], [return _libiconv_version])], [
```

Yes, it seems that link check is sufficient to figure out if iconv library is GNU libiconv. This will expand to something like this:

```c
/* ... confdefs.h ... */
#include <iconv.h>
int main(void) {
  return _libiconv_version
  ;
  return 0;
}
```

For the //IGNORE check, I'm not sure there is any way but to change the check into AC_CACHE_CHECK and set a cache variable of php_cv_something, which can be then set manually when cross-compiling. At some point it could be probably removed in the future because checking for the broken iconv library without the IGNORE support might be obsolete. There is even some error in the php-src C code. In the php-src C code there is a check of `#ifdef ICONV_BROKEN_IGNORE` for about 4 years now and `ICONV_BROKEN_IGNORE` is always defined. Either to a value of 0 or 1. I think we'll need to recheck this a bit further, which part to change and if perhaps this check can be today removed as obsolete.

Reply:
pushed suggested change, looks like the //IGNORE thing is still undecided in glibc... https://sourceware.org/bugzilla/show_bug.cgi?id=13541